api/envoy/config/filter/network/thrift_proxy/v2alpha1/route.proto

```diff
@@ -84,6 +84,11 @@ message RouteAction {
     // assigned to each cluster.
     WeightedCluster weighted_clusters = 2;
   }
+
+  // Optional endpoint metadata match criteria. Only endpoints in the upstream
+  // cluster with metadata matching that set in metadata_match will be
+  // considered. Keys and values should be provided under the "envoy.lb" metadata key. Note that
+  envoy.api.v2.core.Metadata metadata_match = 3;
```

Comment:
I know this got copied from the RDS route.proto, but we should probably make clear that this only applies to the subset load balancer. So like

> Optional endpoint metadata match criteria used by subset load balancer.

Also, there's a dangling "Note that" at the end of the comment.